### ISSUE TITLE
Two-factor authentication

### ISSUE THREAD

**Issue by josephspens** [pwmckenna/generator-travis-ci#8]
I keep failing to pass authorization during the setup process, I believe it's because I have two-factor authentication enabled on my account. Does this generator take that into account, and/or is there a way around it?

**Comment by pwmckenna** [comment]
I'm not really sure...could you describe how you set up your two factor auth so I can take a crack at reproducing the issue?

**Comment by josephspens** [comment]
I followed [Github's tutorial](https://help.github.com/articles/about-two-factor-authentication) when setting up two-factor auth, and found [an article](https://developer.github.com/v3/auth/#working-with-two-factor-authentication) describing extra steps needed for two-factor auth.

Hope this helps.

**Comment by pwmckenna** [comment]
I'll give it a go this evening, and let you know what I find. Worst case scenario, I can walk you through getting your repo set up manually...speaking of which, I should write a blog post describing exactly which steps the generator takes.

**Comment by josephspens** [comment]
Thank you so much, I've been working on this issue on and off for a couple months and got close a few times. I was going to copy your generator template for the travis .yml file and do it manually, I'll go through with that if two-factor auth turns out to be a bigger pain in the ass then expected.

**Comment by pwmckenna** [comment]
egads! don't hesitate to ask for help next time. months is much too long to go without glorious travis-ci-ing!

**Comment by elldritch** [comment]
+1. Generator dies on "Ensure Github Travis App Authorized" even though it's already been authorized and I click "Allow Access" every time it brings the prompt up.

**Comment by pwmckenna** [comment]
on it...turns our there were a couple changes that were breaking the existing functionality altogether...patched that up and released last night. this is the only thing left on my plate.

**Comment by pwmckenna** [comment]
looks like we're going to be in a holding pattern until [this patch](https://github.com/mikedeboer/node-github/pull/114) is merged into the github module. its already been reviewed, so shouldn't be too much longer. once that's done, it'll hopefully be a pretty easy fix to add support for the two factor auth stuff. thanks for the input!

**Comment by pwmckenna** [comment]
@The2ndOne3 @josephspens support for 2FA just landed in master...I've tested it a bunch and it looks good, but I wanted to give you guys a chance to try it out before publishing it if you'd like. Thanks! :beers:

**Comment by elldritch** [comment]
Still dies for me -- I'm prompted to authorise Travis, I accept, and then it takes me to the Travis website where I'm not logged in. Even after logging in on the site, the generator is still stalled at "Ensure Github Travis App Authorized".

**Comment by pwmckenna** [comment]
@The2ndOne3 good catch...didn't fully comprehend how much 2FA affected things. master should handle the travis app authorization step fine now. thanks so much for helping test!

**Comment by josephspens** [comment]
Hmm I'm still getting some errors

```
 (travis) ~/Development/josephspens.github.io 
joseph-> yo travis-ci


   ╔══════════════════════════════════╗
   ║¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤║
   ║¤¤¤¤¤╔══════════════════════╗¤¤¤¤¤║
   ║¤¤¤¤¤║ ╔══════════════════╗ ║¤¤¤¤¤║
   ║¤¤¤¤¤║ ║ ╔═════╗  ╔═════╗ ║ ║¤¤¤¤¤║
   ║¤¤¤¤¤║ ╚═╝ ╔═╗ ║  ║ ╔═╗ ╚═╝ ║¤¤¤¤¤║
   ║¤¤¤¤¤╚═════╝¤║ ║  ║ ║¤╚═════╝¤¤¤¤¤║
   ║¤¤¤¤¤¤¤¤¤¤¤¤¤║ ║  ║ ║¤¤¤¤¤¤¤¤¤¤¤¤¤║
   ║¤¤¤¤¤¤¤¤¤¤¤¤¤║ ║  ║ ║¤¤¤¤¤¤¤¤¤¤¤¤¤║
   ║¤¤¤¤¤¤¤¤¤¤¤¤¤║ ║  ║ ║¤¤¤¤¤¤¤¤¤¤¤¤¤║
   ║¤¤¤¤¤¤¤¤¤¤¤¤¤║ ║  ║ ║¤¤¤¤¤¤¤¤¤¤¤¤¤║
   ║¤¤¤¤¤¤¤¤¤¤¤¤╔╝ ║  ║ ╚╗¤¤¤¤¤¤¤¤¤¤¤¤║
   ║¤¤¤¤¤¤¤¤¤¤¤¤║ ╔╝  ╚╗ ║¤¤¤¤¤¤¤¤¤¤¤¤║
   ║¤¤¤¤¤¤¤¤¤¤¤¤║ ╚════╝ ║¤¤¤¤¤¤¤¤¤¤¤¤║
   ║¤¤¤¤¤¤¤¤¤¤¤¤╚════════╝¤¤¤¤¤¤¤¤¤¤¤¤║
   ║¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤¤║
   ╚══════════════════════════════════╝


[?] What would you like Travis to do for you? travis-ci:gh-pages   (runs `grunt` on Travis-CI and deploys the build to your GitHub Pages branch)
   invoke   travis-ci:gh-pages
√ get repository owner
√ get project name
[?] User & Organization pages build into the master branch. Which branch would you like to develop in? source
√ get source branch
√ get build output branch
[?] GitHub Username: josephspens
[?] GitHub Password: **************
[?] GitHub 2FA One Time Password: 702329
✗ generate github oauth token - {"message":"Bad credentials","documentation_url":"http://developer.github.com/v3"}
✗ get user email - {"message":"Bad credentials","documentation_url":"http://developer.github.com/v3"}
```

**Comment by pwmckenna** [comment]
I sadly can't get this to repo...are you using sms 2FA? I've played around with failed auth a bit and it would seem that the one time password isn't even sent if the password isn't correct, so the fact that you got the code means your credentials were inputed fine...which leaves me with no good theory as to why this is happening.

**Comment by josephspens** [comment]
I'm using Google Authenticator, it's an app which generates keys for 2FA apps, not SMS.

**Comment by aawc** [comment]
+1 Exactly what @josephspens is facing. I'm also using the Android authenticator app.
